internal/run/test_runner.go

```diff
@@ -160,9 +163,12 @@ func (r *Run) Do(ctx context.Context, s *scenarios.Scenarios) (*Result, error) {
 		}
 	}()
 
+	progressCtx, progressCancel := context.WithCancel(ctx)
```

Comment:
What's your thinking RE favouring context as a form of cancellation as opposed to encapsulating it as a method of the progrss runner? I'm fairly indifferent in terms of which I prefer but curious as to your thinking.

Reply:
added a stop func